torch/nested/_internal/ops.py

```diff
@@ -842,15 +857,35 @@ def sum_dim_IntList(func, *args, **kwargs):
     )
     inp = new_kwargs.pop("input")
     assert inp._ragged_idx == 1
-    new_kwargs["dim"], ragged_reduced_away = _wrap_jagged_dims(
+
+    if (
+        0 not in new_kwargs["dim"]
+        and 1 in new_kwargs["dim"]
```

Comment:
as mentioned above, I think we need to take into account NJTs with ragged_idx != 1

Reply:
agreed, and this is actually going to be added in a later diff! for now, as discussed above, i'll refactor these names to be more explicit/clear, and modify the logic here when i add support for NJTs with ragged_idx != 1